File: app/src/main/java/xyz/b515/schedule/api/ZfService.java
package xyz.b515.schedule.api;

        import java.util.Map;

        import io.reactivex.Observable;
        import retrofit2.http.FieldMap;
        import retrofit2.http.FormUrlEncoded;
        import retrofit2.http.GET;
        import retrofit2.http.POST;
        import retrofit2.http.QueryMap;

public interface ZfService {
    @FormUrlEncoded
    @POST("/default_vsso.aspx")
    Observable<String> login(@FieldMap Map<String, String> map);

    @GET("/xskbcx.aspx")
    Observable<String> getSchedule(@QueryMap Map<String, String> map);
}

File: app/src/main/kotlin/xyz/b515/schedule/api/ZfService.kt
package xyz.b515.schedule.api

import io.reactivex.Flowable
import retrofit2.http.*

interface ZfService {
    @FormUrlEncoded
    @POST("/default_vsso.aspx")
    fun login(@FieldMap map: Map<String, String>): Flowable<String>

    @GET("/xskbcx.aspx")
    fun getSchedule(@QueryMap map: Map<String, String>): Flowable<String>
}
